### ISSUE TITLE
Errors with the public demo application

### ISSUE THREAD

**Issue by nguyenLisa** [ABenchM/abm#46]
- [ ] Searching for "demo" in the Public collections page generates a 403 error.
- [x] Downloading the Hermes results generates a 403 error.
- [ ] Downloading the collections generates a 404 error (which is then redirected to the home page).

**Comment by ankur-1989** [comment]
I looked into this issue. So to reproduce this issue. Following are the steps that I did along with my findings -
1. I built two repositories and ran Hermes filter on that.
2. Then I made the collections public to see after logging out of the demo user.
3. Then I tested all the three issues given above on the newly added public collections.
4. So first two are not working reasons given below and third one is working fine for those two collections.
Reason for First Issue - 
 1. Search works fine but in search Alex added one check - whether the searched collections exist in collectionPin table or not with the sessionuser. Since the sessionUser is undefined , so it does not match and return false and hence the error comes.

I have not fixed it yet because I am not fully aware of Pin and Unpin feature and the motto behind this functionality. We can discuss that in meeting on Monday.

Reason for Second Issue:- 

To download the collection it fetches the Hermes instance of that collection. It fails to fetch that instance because only registered users can only fetch those details.
I have fixed this issue. This is working fine now. I have not deployed the changes. I thought you would like to check if the third issue is really there or not on newly added collections. 

I wanted to deploy both the fixes together.

**Comment by nguyenLisa** [comment]
The pin/unpin feature allows users to keep their "favourite" collections somewhere so that they can access them later. So when logged in, users should have an extra section in the Collections page that shows their pinned collections (those collections are always public collections). When logged out, the pinned section is not shown.

In all cases, the search and download functionalities should be independent of whether the user is logged in or not. They should always be able to search for public collections and download them.

Does that answer your question?

About the third issue, if you want to see it, just go on the prod application, and try to download the demo collection. I think that for some reason, the collection does not exist, so a 404 error is generated (and the Apache config redirects it to the home page). This does not happen for the other collections, which we can download correctly.

A fix for that would be to check whether the collection exists, and to generate an error notification if that is not the case.

Also, this should also happen when the user clicks on Download Hermes results and the result file does not exist.

**Comment by ankur-1989** [comment]
I have one more question - Is it like only pinned collections should be visible on Public collection page ?? 
As of now , both pinned and unpinned collections are visible on public collections page .

Solutions possible -
 1. Only pinned collections should be visible on public collections page.
2. or we should not check if it is pinned or unpinned because it does not matter when you are not logged in the application or does it ??

**Comment by nguyenLisa** [comment]
You can only pin collections if they are public.
So when logged out, you should see all public collections. When logged in, you should see your own private collections, all public collections, and the public collections you have pinned because you want to keep track of them for later.

Doest that answer your question?

**Comment by ankur-1989** [comment]
No I meant to say - we see collections pinned or unpinned on the same page when you are logged in. These collections only are visible on public collections page when we are logged out. So my question is - Do we need to show only pinned collections on public collection page or both? 
Second is if we are showing both pinned and unpinned on public collections page then whether we should check if the collection is pinned or not ? I still dont understand the logic of checking if it is pinned when I am logged out.

**Comment by nguyenLisa** [comment]
You shouldn't need to check if the collections are pinned or not when you are logged out.
Pinned collections are just for a particular user, so when that user is logged in. When logged out, only the list of public collections should be shown, whether or not they are pinned.

**Comment by nguyenLisa** [comment]
That's because a server configuration redirects all 404 to the home page. We could remove this configuration but I would rather keep it as is. 
Instead, would it be possible to do a check before generating the page and only allow download of files that exist?

**Comment by ankur-1989** [comment]
Ah ok. However , current application only allows download of files that exist if does not exist it redirects the page back to list of collections ,but if we want to show a notification that file does not exist so to do that either we have to remove that configuration or have a find a different solution.  

I implemented one more solution and tested thats why it took me so much time . In that solution one issue I was facing which works as follows 
- you call a rest end point and check if the file exist or not at the back end and come back with true or false based on file exists or not.  
so here when I store that value in a variable it takes that value in second time.

**Comment by ankur-1989** [comment]
That's why I implemented the other method but this configuration has restricted this solution.

**Comment by nguyenLisa** [comment]
Why not do the check when the page is generated? Then, if the file does not exist, we just don't show the "Download Collection" button. Same for the "Download Hermes results" button.
What do you think?

**Comment by ankur-1989** [comment]
yes this can be done with rest end point approach. i'll make changes and test.

**Comment by ankur-1989** [comment]
Done and tested. Do I only need to make the buttons invisible or need to keep the message as well in place of buttons like File does not exist ?? 
Will deploy the changes after this.

**Comment by nguyenLisa** [comment]
Just greying out the buttons will be fine. If a tooltip could also be added with "The collection does not exist.", it would also be great.

**Comment by ankur-1989** [comment]
Fixed the last part of the issue. Implemented in production. please check and let me know if everything works fine then I will close the ticket.

**Comment by nguyenLisa** [comment]
It works fine, feel free to close the issue. Thanks!
